chia/wallet/wallet_blockchain.py

```diff
@@ -192,7 +192,11 @@ async def get_finished_sync_up_to(self) -> uint32:
     def get_latest_timestamp(self) -> uint64:
         return self._latest_timestamp
 
-    def contains_block(self, header_hash: bytes32) -> bool:
+    def contains_block(self, header_hash: bytes32, height: Optional[uint32] = None) -> bool:
+        """
+        True if we have already added this block to the chain. This may return false for orphan blocks
+        that we have added but no longer keep in memory.
+        """
```

Comment:
I think it's important to document this on the full node side as well. Perhaps this wallet counterpart will be removed or disconnected from the full node eventually

Reply:
but this is no longer true for the full node, after this pr contains block will return true/false according to the height to hash and not the cache